src/lc3_vm/virtual_machine.rs

```diff
@@ -143,108 +158,267 @@ impl VM {
     fn execute(&mut self, opcode: Opcode) -> Result<(), VMError> {
         match opcode {
             Opcode::BR { n, z, p, offset } => {
-                println!("{n}");
-                println!("{z}");
-                println!("{p}");
-                println!("{offset}");
+                let flags_value = self
+                    .get_flags()
+                    .map_err(|err| VMError::Execute(format!("BR {}", err)))?;
+                let offset = sign_extend_9_bits(offset);
+                if n && flags_value == ConditionFlags::NEG.into() {
+                    self.add_to_pc(offset);
+                }
+                if z && flags_value == ConditionFlags::ZRO.into() {
+                    self.add_to_pc(offset);
+                }
+                if p && flags_value == ConditionFlags::POS.into() {
+                    self.add_to_pc(offset);
+                }
```

Comment:
I think because of the specification the condition flag can only be one thing at the time.

However if this were not the case then if `n` and `z` and `p` were all true then the offset would be added three times, I think it's less error prone if you use `else if` here to make 100% sure we only execute one branch.

Reply:
I think that they should be mutually exclusive but added it just to make sure.